### PR TITLE
upgrade to go 1.20 and newer protobuf builder to reduce v50 changeset

### PR DESCRIPTION
**Pull request by faddat**
This PR uses go 1.20 and the latest protobuf builder so that the changeset for 
v50 is smaller.  This is the same intent as the earlier thelper PR, basically to 
reduce the total number of changes that will be present in the v50 pr.

### CODE REVIEW

**Review by crodriguezvega — CHANGES_REQUESTED** [review]
Thank you, @faddat! I think it would be nice to take out the bump to SDK v0.47.3 out of this PR (since you have that covered in #3810) and then the scope of this PR will very nicely be scope to bumping go version and protobuf builder version. :)

If these are changes required for SDK v0.50. Should we target this PR to the [feature branch](https://github.com/cosmos/ibc-go/tree/feat/upgrade-sdk-v0.50)?

**Comment by faddat** [comment]
Hey Carlos, I'm very happy to do that. I will take care of that right now.

**Comment by faddat** [comment]
...btw I do have a question -- I could not really figure out why the capabilities module does not build here, do you have any ideas?

**Comment by crodriguezvega** [comment]
> I could not really figure out why the capabilities module does not build here, do you have any ideas?

Looks like it's building now, right?

**Review by crodriguezvega — APPROVED** [review]
Looks good; thank you, @faddat!

File: .github/workflows/capability.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v4
         with:
-          go-version: 1.19
+          go-version: "1.20"
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply .github/workflows/capability.yml:20]: apparently they [recommend single quotes](https://github.com/actions/setup-go#v3) due to some yaml parsing but I don't think it'll make too much of a difference

**Review by damiannolan — APPROVED** [review]
Thanks @faddat, lgtm!

**Review by DimitrisJim — COMMENTED** [review]
:rocket:

**Review by DimitrisJim — APPROVED** [review]

**Comment by faddat** [comment]
I'll make it single quotes because following the recommendations is great :)

**Comment by colin-axner** [comment]
Was this commit https://github.com/cosmos/ibc-go/pull/3811/commits/dd3b04b51dc7b8cb700cfdf7c41fcd4bb05d8c10 supposed to turn the quotes into single quotes or is the intention to do it in a followup commit?

**Comment by faddat** [comment]
Hey Colin, yeah that was definitely supposed to be single quotes and I think that I should just revert that commit, I'm not sure how it ended up that way

**Comment by faddat** [comment]
So what ended up being necessary was just reconfiguring my editor to use single quotes in YAML by default. 

When I did the flub commit before, it was because my editor was changing the singles to doubles when I saved the file.

**Review by colin-axner — APPROVED** [review]

**Pull request merged**